content/en/smart-contracts/fundamentals/programmatic-storage/index.md

```diff
@@ -0,0 +1,60 @@
+
+## Programmatic storage
+
+The Filecoin virtual machine (FVM) is not designed to directly interact with data on Filecoin. Instead, it operates on blockchain state data, which includes metadata about stored data. The FVM's role is to compute over this state data and create logic around the storage and retrieval process. Storage and retrieval providers on the Filecoin network handle the actual process of storing and retrieving data.
+
+Develoeprs can use FVM actors (also called smart-contracts) to implement various use cases related to data storage and retrieval on the Filecoin network. For instance, they can automate the process of renewing storage deals, replicate data across multiple storage providers for redundancy, or implement access control rules for data retrieval.
+
+The FVM also includes an Ethereum Virtual Machine (EVM) runtime, enabling developers to deploy existing EVM-based smart contracts onto the FVM with little to no modifications. If you already have a smart contract written in Solidity, you can deploy it to Filecoin and have it interact with the Filecoin storage system.
+
+This process for programmatically storing data on Filecoin differs depending on how much data you need to store. Anything less than 4 GiB of data is considered _small_, while anything 4 GiB or above is considered _large_. Lets cover the process for both _small_ and _large_ storage deals:
+
+### Small-scale storage
+
+Storage deals for less than 4 GiB of data are considered _small_ on the Filecoin network. This is because storage providers use sectors of either 32 GiB or 64 GiB. Because of these comparably large sector sizes, SPs are incentivized to accept storage deals closer to this cap size. An SP will receive more FIL rewards for storing a 31 GiB file rather than a 200 KiB file. Clients will have difficulty finding a storage provider to accept deals for small amounts of data.
+
+To get around this problem, these developers can _aggregate_ their data into bundles. The deal-aggregation process bundles a collection of small storage deals with many others to create one large deal. 
+
+Examples of deal aggregators developers can use include: 
+
+- [Lighthouse.storage](https://www.lighthouse.storage/documentation)
+- [FDT's Edge-UR tool](https://github.com/application-research/edge-ur/blob/car-gen/docs/aggregation.md). 
+
+### Large-scale storage
+
+Storage deals for more than 4 GiB don't suffer from the limitations of small-scale storage deals. Large-scale deals can be processed and sent to an SP directly without aggregating data. Client contracts are used to automatically find an SP to store your data. The process goes something like this:
```

Comment:
Strongly suggest we link out the client contract process to the tutorial for consistency and easy maintenance: https://docs.filecoin.io/smart-contracts/developing-contracts/client-contract-tutorial/

This paragraph can look like "Storage deals for more than 4 GiB don't suffer from the limitations of small-scale storage deals. Large-scale deals can be processed and sent to an SP directly without aggregating data. Client contracts are used to automatically find an SP to store your data. Note that there are other considerations for an SP to decide to pick up your data deal proposals from the client contract. See the tutorial here [hyperlink to https://docs.filecoin.io/smart-contracts/developing-contracts/client-contract-tutorial/].

Reply:
Done.